scripts/tools/generate_esp32_chip_factory_bin.py

```diff
@@ -164,15 +308,31 @@ def any_base_int(s): return int(s, 0)
                         help='The path to the PAI certificate in der format')
     parser.add_argument('--cd', type=str, required=True,
                         help='The path to the certificate declaration der format')
+
+    # These will be used by DeviceInstanceInfoProvider
+    parser.add_argument('--vendor-id', type=any_base_int, required=False, help='Vendor id')
+    parser.add_argument('--vendor-name', type=str, required=False, help='Vendor name')
+    parser.add_argument('--product-id', type=any_base_int, required=False, help='Product id')
+    parser.add_argument('--product-name', type=str, required=False, help='Product name')
+    parser.add_argument('--hw-ver', type=any_base_int, required=False, help='Hardware version')
+    parser.add_argument('--hw-ver-str', type=str, required=False, help='Hardware version string')
+    parser.add_argument('--mfg-date', type=str, required=False, help='Manufacturing date in format YYYY-MM-DD')
+    parser.add_argument('--serial-num', type=str, required=False, help='Serial number')
+    parser.add_argument('--unique-id', type=str, required=False,
+                        help='128-bt unique identifier, provide 32-byte hex string, e.g. "1234567890abcdef1234567890abcdef"')
+
     parser.add_argument('-s', '--size', type=any_base_int, required=False, default=0x6000,
                         help='The size of the partition.bin, default: 0x6000')
 
     args = parser.parse_args()
     validate_args(args)
     check_tools_exists()
     spake2p_params = gen_spake2p_params(args.passcode)
-    generate_nvs_bin(args, spake2p_params)
+    populate_factory_data(args, spake2p_params)
+    gen_raw_ec_keypair_from_der(args.dac_key, FACTORY_DATA['dac-pub-key']['value'], FACTORY_DATA['dac-key']['value'])
```

Comment:
Is it possible to encrypt the DAC key using a device-specific key, or is this already done by the NVS layer, or does the device attestation private key appear directly in the flash?